### ISSUE TITLE
N+1 queries when batch loading has many association

### ISSUE THREAD

**Issue by tjvc** [exAspArk/batch-loader#59]
Nice gem - thanks!

This isn't an issue per se - I already have a working solution based on the second example from this section in the README: https://github.com/exAspArk/batch-loader#loading-multiple-items.

I'm curious though as to why the following approach does not prevent an N+1 query.

Given these models:

```ruby
class Post < ApplicationRecord
  has_many :comments
end

class Comment < ApplicationRecord
  belongs_to :post
end
```

And a GraphQL schema with a `postsConnection` on the root query type and a `commentsConnection` on the `post` type.

The following seems to trigger an N+1 query for the comments association on each post when used to resolve the `commentsConnection` for each `post`:

```ruby
BatchLoader::GraphQL.for(post.id).batch(default_value: []) do |post_ids, loader|
  Post.
    where(id: post_ids).
    includes(:comments).
    each { |post| loader.call(post.id, post.comments) }
end
```

**Comment by exAspArk** [comment]
Hey @tjvc,

Interesting. A few questions:

* Do you have `use BatchLoader::GraphQL` in your schema?
* What `BatchLoader` and `GraphQL` versions do you use?
* Could you show the smallest reproducible code sample? (e.g. a repo)

**Comment by tjvc** [comment]
Hey @exAspArk. Thanks for the suggestion to try reproducing this in a smaller repo. I tried in a new application and couldn't. Digging into this a bit more, I discovered that it was related to our Pundit integration with GraphQL::Pro, and specifically a feature of our implementation (we were chaining scopes onto authorised relations, which triggered fresh queries). So I'll close this now as it's definitely not an issue with BatchLoader, and unlikely to be something experienced by other users. Thanks for your time!
